src/directory/file_watcher.rs

```diff
@@ -104,35 +108,37 @@ mod tests {
         let (tx, rx) = crossbeam::channel::unbounded();
         let timeout = Duration::from_millis(100);
 
-        {
-            let watcher = FileWatcher::new(&tmp_file);
+        let watcher = FileWatcher::new(&tmp_file);
 
-            state = watcher.state.clone();
-            assert_eq!(state.load(Ordering::SeqCst), 0);
+        state = watcher.state.clone();
+        assert_eq!(state.load(Ordering::SeqCst), 0);
 
-            let counter_clone = counter.clone();
+        let counter_clone = counter.clone();
 
-            _handle = watcher.watch(WatchCallback::new(move || {
-                let val = counter_clone.fetch_add(1, Ordering::SeqCst);
-                tx.send(val + 1).unwrap();
-            }));
+        _handle = watcher.watch(WatchCallback::new(move || {
+            let val = counter_clone.fetch_add(1, Ordering::SeqCst);
+            tx.send(val + 1).unwrap();
+        }));
 
-            assert_eq!(counter.load(Ordering::SeqCst), 0);
-            assert_eq!(state.load(Ordering::SeqCst), 1);
+        assert_eq!(counter.load(Ordering::SeqCst), 0);
+        assert_eq!(state.load(Ordering::SeqCst), 1);
 
-            fs::write(&tmp_file, b"foo").unwrap();
-            assert_eq!(rx.recv_timeout(timeout), Ok(1));
+        atomic_write(&tmp_file, b"foo")?;
+        assert_eq!(rx.recv_timeout(timeout), Ok(1));
 
-            fs::write(&tmp_file, b"foo").unwrap();
-            assert!(rx.recv_timeout(timeout).is_err());
+        atomic_write(&tmp_file, b"foo")?;
+        assert!(rx.recv_timeout(timeout).is_err());
 
-            fs::write(&tmp_file, b"bar").unwrap();
-            assert_eq!(rx.recv_timeout(timeout), Ok(2));
-        }
+        atomic_write(&tmp_file, b"bar")?;
+        assert_eq!(rx.recv_timeout(timeout), Ok(2));
 
-        fs::write(&tmp_file, b"qux").unwrap();
+        mem::drop(watcher);
```

Comment:
Now we're testing that when we drop the handle, the callback list becomes empty, and broadcast is no-op. I guess we should write two tests. I can work on that.